File: tests/integration/deleted_with/nodejs/random.ts
// Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

import * as codeinfra from "@codeinfra/codeinfra";

interface RandomArgs {
    length: codeinfra.Input<number>;
}

export class Random extends codeinfra.CustomResource {
    public readonly length!: codeinfra.Output<number>;
    public readonly result!: codeinfra.Output<string>;
    constructor(name: string, args: RandomArgs, opts?: codeinfra.CustomResourceOptions) {
        super("testprovider:index:Random", name, args, opts);
    }
}

File: tests/integration/transformations/python/simple/random_.py
# Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

from typing import Optional

import codeinfra

class Random(codeinfra.CustomResource):
    def __init__(self,
                 resource_name: str,
                 length: codeinfra.Input[int],
                 prefix: Optional[codeinfra.Input[str]] = None,
                 opts: Optional[codeinfra.ResourceOptions] = None):
        props = {
            "length": length,
            "result": None,
            "prefix": prefix,
        }
        super().__init__("testprovider:index:Random", resource_name, props, opts)

    @property
    @codeinfra.getter
    def length(self) -> codeinfra.Output[int]:
        return codeinfra.get(self, "length")

    @property
    @codeinfra.getter
    def result(self) -> codeinfra.Output[str]:
        return codeinfra.get(self, "result")
